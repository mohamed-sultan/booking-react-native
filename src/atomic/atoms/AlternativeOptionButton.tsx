import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import ArrowUp from './ArrowRight';
import { useThemeColors } from '@store/useTheme';

const AlternativeOptionButton = ({onPress}: any) => {
  const colors = useThemeColors();
  const styles = getStyles(colors);
  return (
    <TouchableOpacity
       accessibilityRole="button"
      style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>I'd prefer to speak with a GP</Text>
      <ArrowUp size={18} />
    </TouchableOpacity>
  );
};

export default React.memo(AlternativeOptionButton);

const getStyles = (colors: any) => StyleSheet.create({
  button: {
    backgroundColor: colors.LightBackground,
    borderRadius: 8,
    paddingVertical: 12,
    marginBottom: 32,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    gap: 4,
    opacity: 0.7,
  },
  buttonText: {
    fontSize: 14,
    textAlign: 'center',
    color: colors.PrimaryText,
  },
});

