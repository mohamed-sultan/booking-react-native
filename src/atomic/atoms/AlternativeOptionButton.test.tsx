import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AlternativeOptionButton from './AlternativeOptionButton';

// ✅ Mock useThemeColors
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    LightBackground: '#f0f0f0',
    PrimaryText: '#333',
  }),
}));

// ✅ Mock ArrowRight icon component
jest.mock('./ArrowRight', () => () => {
  return <></>; // Return empty fragment as a placeholder
});

describe('AlternativeOptionButton', () => {
  it('renders with correct text', () => {
    const { getByText } = render(<AlternativeOptionButton onPress={jest.fn()} />);
    expect(getByText("I'd prefer to speak with a GP")).toBeTruthy();
  });

  it('calls onPress when tapped', () => {
    const onPressMock = jest.fn();
    const { getByRole } = render(<AlternativeOptionButton onPress={onPressMock} />);
    const button = getByRole('button');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const tree = render(<AlternativeOptionButton onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
