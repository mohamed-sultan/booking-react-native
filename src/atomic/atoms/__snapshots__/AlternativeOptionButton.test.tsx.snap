// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlternativeOptionButton matches the snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f0f0f0",
      "borderRadius": 8,
      "flexDirection": "row",
      "gap": 4,
      "justifyContent": "center",
      "marginBottom": 32,
      "opacity": 0.7,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    I'd prefer to speak with a GP
  </Text>
</View>
`;
